### DISCUSSION TITLE
Fastlane download and upload of ITC app review movies

### DISCUSSION THREAD

**Discussion by hixfield** [fastlane/fastlane#18079] [Q&A]
Is the download and upload of ITC app preview movies supported?
I find some mentioned scattered around but nothing in the documentation...

**Comment by dinsen** [comment]
Hi @hixfield 
The App Store Connect endpoints for download and upload of previews are added in Spaceship. But not sure if there are any actions that can do it.

You should be able to do something like this:
```ruby
app = Spaceship::ConnectAPI::App.find("com.my.bundle.id")
app_version = app.get_edit_app_store_version
app_version_localication = app_version.get_app_store_version_localizations.first
app_preview_sets = app_version_localication.get_app_preview_sets
```
I have not tested if this actually works, but hopefully this can get you in the right direction 😄

**Reply by TerryFranguiadakis** [reply]
This doesn't work: this is the error I get: 
 The given operation is not allowed - The resource 'appPreviewSets' does not allow 'GET_COLLECTION'. Allowed operations are: GET_INSTANCE, CREATE, DELETE
